examples/40_advanced/example_text_preprocessing.py

```diff
@@ -22,67 +22,54 @@
 ############################################################################
 # Data Loading
 # ============
+cats = ["comp.sys.ibm.pc.hardware", "rec.sport.baseball"]
+X_train, y_train = fetch_20newsgroups(
+    subset="train",  # select train set
+    shuffle=True,  # shuffle the data set for unbiased validation results
+    random_state=42,  # set a random seed for reproducibility
+    categories=cats,  # select only 2 out of 20 labels
+    return_X_y=True,  # 20NG dataset consists of 2 columns X: the text data, y: the label
+)  # load this two columns separately as numpy array
+
+X_test, y_test = fetch_20newsgroups(
+    subset="test",  # select test set for unbiased evaluation
+    categories=cats,  # select only 2 out of 20 labels
+    return_X_y=True,  # 20NG dataset consists of 2 columns X: the text data, y: the label
+)  # load this two columns separately as numpy array
 
-newsgroups_train = fetch_20newsgroups(subset="train", random_state=42, shuffle=True)
-newsgroups_test = fetch_20newsgroups(subset="test")
-
-# load train data
-df_train = pd.DataFrame({"X": [], "y": []})
-
-for idx, (text, target) in enumerate(
-    zip(newsgroups_train.data, newsgroups_train.target)
-):
-    df_train = pd.concat(
-        [
-            df_train,
-            pd.DataFrame(
-                {"X": text, "y": newsgroups_train.target_names[target]}, index=[idx]
-            ),
-        ]
-    )
-
-# explicitly label text column as string
-X_train = df_train.astype({"X": "string", "y": "category"})
-
-# show all 20 labels
-print(list(newsgroups_train.target_names))
-
-# reduce the example to only 5 labels
-five_newsgroups_labels = list(newsgroups_train.target_names)[:5]
-
-X_train = X_train[~X_train["y"].isin(five_newsgroups_labels)]
-y_train = X_train.pop("y")
-
-# load test data
-df_test = pd.DataFrame({"X": [], "y": []})
-
-for idx, (text, target) in enumerate(zip(newsgroups_test.data, newsgroups_test.target)):
-    df_test = pd.concat(
-        [
-            df_train,
-            pd.DataFrame(
-                {"X": text, "y": newsgroups_train.target_names[int(target)]},
-                index=[idx],
-            ),
-        ]
-    )
-
-# explicitly label text column as string
-X_test = df_test.astype({"X": "string", "y": "category"})
-X_test = X_test[~X_test["y"].isin(five_newsgroups_labels)]
-y_test = X_test.pop("y")
+############################################################################
+# Creating a pandas dataframe
+# ===========================
+# Both categorical and text features are often strings. Python Pandas stores python stings
+# in the generic `object` type. Please ensure that the correct
+# `dtype <https://pandas.pydata.org/docs/user_guide/basics.html#dtypes>` is applied to the correct
```

Comment:
```suggestion
# `dtype <https://pandas.pydata.org/docs/user_guide/basics.html#dtypes>_` is applied to the correct
```

Reply:
Or must the underscore go after the backtick :thinking:

Reply:
example_debug_logging.py makes it like this
``SMAC documentation <https://github.com/automl/SMAC3>`_`